tools/ports/__init__.py

```diff
@@ -42,6 +46,14 @@ def read_ports():
       port.linker_setup = lambda x, y: 0
     if not hasattr(port, 'deps'):
       port.deps = []
+    # port variants (default: no variants)
+    if not hasattr(port, 'get_variants'):
+      port.get_variants = lambda: {}
```

Comment:
How about making this a static like `deps`?

Reply:
Sounds good to me!